AudioKit/Common/Internals/Helpers/ExceptionCatcher.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  ExceptionCatcher.swift
+//  AudioKit
+//
+//  Created by Stéphane Peter, revision history on Github.
+//  Copyright © 2018 AudioKit. All rights reserved.
+//
+
+import Foundation
+
+extension NSException: Error
+{
+    public var domain: String { return "io.audiokit.SwiftExceptionCatcher" }
+    public var code: Int { return 0 }
+}
+
+public func AKTry(_ operation: @escaping (() throws -> Void),
```

Comment:
As it is now, operation should not take a throwing closure as any error caught will be ignored.

Reply:
I'm not sure I follow. There is a catch section that assigns the caught exception, which is in turn thrown back as a Swift exception. How is it getting lost?

Reply:
```error = ex as? NSException``` If ex is a Swift error (not a NSException),  error will be nil and no exception will be thrown.

Reply:
OK, though I think the declaration above adds Swift Error protocol conformance to NSException. That's not my original code though, so I should probably make sure it works.

Reply:
Maybe the error var should just be an Error? rather than NSException?, then you could just throw that.